tests/basic/shared_nfsv4_test.go

```diff
@@ -0,0 +1,173 @@
+						nodesNonReplica = append(nodesNonReplica, node)
+					}
+				}
+				for _, node := range nodesNonReplica {
+					Inst().S.DisableSchedulingOnNode(node)
+				}
+
+			})
+
+			// scale down and then scale up the app, so that pods are only scheduled on replica nodes
+			Step(fmt.Sprintf("scale down app: %s to 0 ", ctx.App.Key), func() {
+				applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for name := range applicationScaleUpMap {
+					applicationScaleUpMap[name] = int32(0)
+				}
+				err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			Step(fmt.Sprintf("scale up app: %s to 2, and re-enable scheduling on all nodes", ctx.App.Key), func() {
+				applicationScaleUpMap, err := Inst().S.GetScaleFactorMap(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for name := range applicationScaleUpMap {
+					applicationScaleUpMap[name] = int32(2)
+				}
+				err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+				Expect(err).NotTo(HaveOccurred())
+				ValidateApplications(contexts)
+				for _, node := range nodesNonReplica {
+					Inst().S.EnableSchedulingOnNode(node)
+				}
+			})
+
+			Step("fail over nfs server, and make sure the pod on server gets restarted", func() {
+				oldServer, err := Inst().V.GetNodeForVolume(volume, defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("old nfs server %v [%v]", oldServer.SchedulerNodeName, oldServer.Addresses[0])
+				pods, err := core.Instance().GetPodsUsingPV(volume.ID)
+				Expect(err).NotTo(HaveOccurred())
+				var oldPodOnOldServer corev1.Pod
+				for _, pod := range pods {
+					if pod.Spec.NodeName == oldServer.Name {
+						oldPodOnOldServer = pod
+					}
+				}
+				// make sure there is a pod running on the old nfs server
+				Expect(&oldPodOnOldServer).NotTo(BeNil())
+				logrus.Infof("pod on old server %v, creation time %v", oldPodOnOldServer.Name, oldPodOnOldServer.CreationTimestamp)
+
+				timestampBeforeFailOver := time.Now()
+				err = Inst().V.StopDriver([]node.Node{*oldServer}, false, nil)
+				Expect(err).NotTo(HaveOccurred())
+				err = Inst().V.WaitDriverDownOnNode(*oldServer)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("stopped px on nfs server node %v [%v]", oldServer.SchedulerNodeName, oldServer.Addresses[0])
+
+				var newServer *node.Node
+
+				for i := 0; i < 10; i++ {
+					server, err := Inst().V.GetNodeForVolume(volume, defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+					if server.Id != oldServer.Id {
+						logrus.Infof("nfs server failed over, new nfs server is %s [%s]", server.SchedulerNodeName, server.Addresses[0])
+						newServer = server
+						break
+					}
+					time.Sleep(oneMinute)
+				}
+				// make sure nfs server failed over
+				Expect(newServer).NotTo(BeNil())
+				logrus.Infof("new nfs server is %v", newServer)
```

Comment:
should we log only the relevant info about the new server such as name, IP and maybe id?